cmd/antrea-agent/util_test.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"net"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetAvailableNodePortAddresses(t *testing.T) {
+	testCases := []struct {
+		name                        string
+		nodePortAddressesFromConfig []string
+		excludeDevices              []string
+		expectedipv4                []net.IP
+		expectedipv6                []net.IP
+	}{
+		{
+			name:                        "empty nodePortAddressesFromConfig",
+			nodePortAddressesFromConfig: []string{},
+			excludeDevices:              []string{"antrea-egress0", "antrea-ingress0", "kube-ipvs0", "antrea-gw0"},
+			expectedipv4:                []net.IP{net.IP{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xff, 0xff, 0x7f, 0x0, 0x0, 0x1}, net.IP{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xff, 0xff, 0xc0, 0xa8, 0xe1, 0xea}, net.IP{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xff, 0xff, 0xa, 0x1e, 0x68, 0x48}},
+			expectedipv6:                []net.IP{net.IP{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1}, net.IP{0x24, 0x9, 0x40, 0x71, 0x4d, 0x1e, 0x91, 0x43, 0x18, 0x48, 0xec, 0xb6, 0xbf, 0x56, 0xf, 0x8}, net.IP{0x24, 0x9, 0x40, 0x71, 0x4d, 0x1e, 0x91, 0x43, 0x71, 0x24, 0xa0, 0x20, 0x5c, 0x3e, 0x49, 0xa9}},
```

Comment:
1. Apparently this can only succeed on your own host, as the test failure shows. You need to mock `util.GetAllNodeAddresses` to return specific values to test the target function.
2. Use `net.ParseIP(...)` to make the expected value readable.